dashboard/app/views/levels/_level_group.haml

```diff
@@ -27,15 +27,18 @@
 
     -# Show any text from an external level.
 
-    -# Fall back to retrieving the the last_attempt from the parsed levelgroup last_attempt
-    - sublevel_last_attempt = parsed[level.id.to_s].try(:[], 'result')
-    - if @user && current_user && @user != current_user
-      -# load other user's solution for teachers viewing their students' solution
-      - sublevel_last_attempt = @user.last_attempt(level).try(:level_source).try(:data)
-    - elsif current_user
-      -# load user's previous attempt at this puzzle.
-      - sublevel_last_attempt = current_user.last_attempt(level).try(:level_source).try(:data)
-
+    -# Show last attempted answer, unless this is a teacher looking at a student's attempt
+    -# at an anonymous survey.  In that case it's very important that we do not show the
+    -# last attempt.
```

Comment:
I got a fair way on this, but it's tricky because @last_attempt for the containing level (which is parsed in the fallback case) is generated in ScriptLevelsController and then this new helper function would need to take that into account, or regenerate it separately, or we should actually be generating all the last_attempts in the same place and unifying these codepaths.  Punting right now, but noting for the backlog.